src/main/java/com/fundynamic/d2tm/game/entities/units/Unit.java

```diff
@@ -491,4 +488,49 @@ public Entity getEntityToAttack() {
     public Vector2D getDimensions() {
         return Vector2D.create(TILE_SIZE, TILE_SIZE);
     }
+
+    public void harvest(Coordinate target) {
+        moveTo(target);
+    }
+
+    private void setState(UnitState state) {
```

Comment:
It seems to me like we're guided to remove all logic from `Unit#update` to the relevant state machine classes. Checks like :

```
if (hitPointBasedDestructibility.hasDied()) {
  die();
}
```

should be moved as well (using a default implementation for `State#Update`?). 

I think we need to strive state-changes to only be triggered by states (instead of external factor). Though we cannot prevent this in all cases (like mouse-click to attack). What do you think?